### DISCUSSION TITLE
Is there a better way to fix broken imports?

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#11534] [Libraries & Drivers]
I really like my repl. The interactive edit, run, rinse & repeat is great. 

But I'm struggling with library files. At the moment when an import crashes I have to make an edit on the file, download it to the esp32, run the main.py that calls it, see if things have improved, altogether a more tedious rinse & repeat. Just wondering if anybody knows a better way?

**Comment by davefes** [comment]
rshell on Linux and modify it on /pyboard

**Comment by jimmo** [comment]
There are some little tweaks you can do to improve this workflow (e.g. you can use the `--inject-code` or `--inject-file` options to REPL to automate the interaction at the REPL to start main.py).

`mpremote mount` -- https://docs.micropython.org/en/latest/reference/mpremote.html -- is designed specifically to help with the case of working on a file locally and having it synced automatically to the device. (No need to disconnect the REPL).

In general I try to make it so most of my development cycle is done via `mpremote run`. For example, if you're working on a bug in lib/foo.py, rather than uploading a new lib/foo.py and then running main.py again to test it, write a simple bit of code at the end of lib/foo.py that demonstrates teh bug/feature, and just `mpremote run local/path/to/lib/foo.py` instead.

(Or where possible, using the unix port and get the code working on my PC first instead, e.g. with stubbed/mocked out hardware APIs).

**Reply by kjm1102** [reply]
thoughtful reply as ever Jim, although I might have to swat up on your tweaks before I can use them.
